packages/ui/src/ButtonVariants.tsx

```diff
@@ -76,26 +105,35 @@ export const ButtonBase = styled(ButtonComponent)`
     pointer-events: none;
     opacity: 0.5;
   }
+
+  ${({ icon, size }) =>
+    !!icon &&
+    `
+  pointer-events: none;
```

Comment:
how are we gonna have hover states like this:
![image](https://user-images.githubusercontent.com/22115870/175349335-68c3bcad-a44d-4314-983c-095304cfe329.png)
if we have `pointer-events: none`?